packages/yourloops/components/alarm/alarms-content-configuration.hook.tsx

```diff
@@ -0,0 +1,234 @@
+}
+
+export const MIN_HIGH_BG = 140
+export const MAX_HIGH_BG = 250
+export const MIN_VERY_LOW_BG = 40
+export const MAX_VERY_LOW_BG = 90
+export const MIN_LOW_BG = 50
+export const MAX_LOW_BG = 100
+export const PERCENTAGES = [...new Array(21)]
+  .map((_each, index) => `${index * 5}%`).slice(1, 21)
+
+const useAlarmsContentConfiguration = ({ monitoring, saveInProgress, onSave, patient }: AlarmsContentConfigurationHookProps): AlarmsContentConfigurationHookReturn => {
+  const bgUnit = monitoring?.parameters?.bgUnit ?? UNITS_TYPE.MGDL
+  const isInvalidPercentage = (value: number): boolean => {
+    return !PERCENTAGES.includes(`${value}%`)
+  }
+  const isError = (value: number, lowValue: number, highValue: number): boolean => {
+    const isInRange: boolean = value >= lowValue && value <= highValue
+    const isInteger: boolean = bgUnit === UNITS_TYPE.MGDL && Number.isInteger(value)
+    const isFloat: boolean = bgUnit === UNITS_TYPE.MMOLL && value % 1 !== 0
+    return !(isInRange && (isInteger || isFloat))
+  }
+  const teamHook = useTeam()
+  const [highBg, setHighBg] = useState<ValueErrorPair>({
+    value: monitoring?.parameters?.highBg,
+    error: !monitoring?.parameters?.highBg || isError(monitoring?.parameters?.highBg, MIN_HIGH_BG, MAX_HIGH_BG)
+  })
+  const [veryLowBg, setVeryLowBg] = useState<ValueErrorPair>({
+    value: monitoring?.parameters?.veryLowBg,
+    error: !monitoring?.parameters?.veryLowBg || isError(monitoring?.parameters?.veryLowBg, MIN_VERY_LOW_BG, MAX_VERY_LOW_BG)
+  })
+  const [lowBg, setLowBg] = useState<ValueErrorPair>({
+    value: monitoring?.parameters?.lowBg,
+    error: !monitoring?.parameters?.lowBg || isError(monitoring?.parameters?.lowBg, MIN_LOW_BG, MAX_LOW_BG)
+  })
+  const [nonDataTxThreshold, setNonDataTxThreshold] = useState<ValueErrorPair>(
+    {
+      value: monitoring?.parameters?.nonDataTxThreshold,
+      error: monitoring?.parameters?.nonDataTxThreshold === undefined || isInvalidPercentage(monitoring.parameters.nonDataTxThreshold)
+    })
+  const [outOfRangeThreshold, setOutOfRangeThreshold] = useState<ValueErrorPair>(
+    {
+      value: monitoring?.parameters?.outOfRangeThreshold,
+      error: monitoring?.parameters?.outOfRangeThreshold === undefined || isInvalidPercentage(monitoring.parameters.outOfRangeThreshold)
+    })
+  const [hypoThreshold, setHypoThreshold] = useState<ValueErrorPair>(
+    {
+      value: monitoring?.parameters?.hypoThreshold,
+      error: monitoring?.parameters?.hypoThreshold === undefined || isInvalidPercentage(monitoring.parameters.hypoThreshold)
+    })
+  const saveButtonDisabled = useMemo(() => {
+    return lowBg.error ||
+      highBg.error ||
+      veryLowBg.error ||
+      outOfRangeThreshold.error ||
+      hypoThreshold.error ||
+      nonDataTxThreshold.error ||
+      saveInProgress
+  }, [highBg.error, hypoThreshold.error, lowBg.error, nonDataTxThreshold.error, outOfRangeThreshold.error, saveInProgress, veryLowBg.error])
+  const onChange = (
```

Comment:
This function does not depend on any state or hook and can exported from an util file (could be named `alarm-content-configuration.utils.tsx`)

Reply:
Because of getErrorMessageValue now I can't export it